react/features/prejoin/actions.js

```diff
@@ -58,6 +99,124 @@ export function addPrejoinContentSharingTrack(value: Object) {
     };
 }
 
+/**
+ * Polls for status change after dial out.
+ * Changes dialog message based on response, closes the dialog if there is an error,
+ * joins the meeting when CONNECTED.
+ *
+ * @param {string} reqId - The request id used to correlate the dial out request with this one.
+ * @param {Function} onSuccess - Success handler.
+ * @param {Function} onFail - Fail handler.
+ * @param {number} count - The number of retried calls. When it hits STATUS_REQ_CAP it should no longer make requests.
+ * @returns {Function}
+ */
+function pollForStatus(
+        reqId: string,
+        onSuccess: Function,
+        onFail: Function,
+        count = 0) {
+    return async function(dispatch: Function, getState: Function) {
+        try {
+            if (!isJoinByPhoneDialogVisible(getState())) {
+                return;
+            }
+
+            const res = await executeDialOutStatusRequest(getDialOutStatusUrl(getState()), reqId);
+
+            switch (res) {
+            case DIAL_OUT_STATUS.INITIATED:
+            case DIAL_OUT_STATUS.RINGING: {
+                dispatch(setDialOutStatus(dialOutStatusToKeyMap[res]));
+
+                if (count < STATUS_REQ_CAP) {
+                    return setTimeout(() => {
+                        dispatch(pollForStatus(reqId, onSuccess, onFail, count + 1));
+                    }, STATUS_REQ_FREQUENCY);
+                }
+
+                return onFail();
+            }
+
+            case DIAL_OUT_STATUS.CONNECTED: {
+                return onSuccess();
+            }
+
+            case DIAL_OUT_STATUS.DISCONNECTED: {
+                dispatch(showErrorNotification({
+                    titleKey: 'prejoin.errorDialOutDisconnected'
+                }));
+
+                return onFail();
+            }
+
+            case DIAL_OUT_STATUS.FAILED: {
+                dispatch(showErrorNotification({
+                    titleKey: 'prejoin.errorDialOutFailed'
+                }));
+
+                return onFail();
+            }
+            }
+        } catch (err) {
+            dispatch(showErrorNotification({
+                titleKey: 'prejoin.errorDialOutStatus'
+            }));
+            logger.log('Error getting dial out status', err);
+            onFail();
+        }
+    };
+}
+
+
+/**
+ * Action used so the meeting calls the user.
```

Comment:
Can yo please rephrase this description? It's a bit hard to understand

Reply:
Done.